src/Service.Host/client/src/components/parts/Part.js

```diff
@@ -34,15 +34,206 @@ function Part({
+                    };
+                }
+                if (action.fieldName === 'accountingCompany') {
+                    const updated =
+                        action.payload === 'RECORDS'
+                            ? {
+                                  ...state.part,
+                                  accountingCompany: action.payload,
+                                  paretoCode: 'R',
+                                  bomType: 'C',
+                                  linnProduced: 'N',
+                                  qcOnReceipt: 'N'
+                              }
+                            : { ...state.part, accountingCompany: action.payload, paretoCode: 'U' };
+                    return {
+                        ...state,
+                        part: updated
+                    };
+                }
+                if (action.fieldName === 'sernosSequenceName') {
+                    return {
+                        ...state,
+                        part: {
+                            ...state.part,
+                            sernosSequenceName: action.payload.name,
+                            sernosSequenceDescription: action.payload.description
+                        }
+                    };
+                }
+                if (action.fieldName === 'linnProduced') {
+                    const linnProduced = action.payload === 'Y';
+                    if (linnProduced) {
+                        return {
+                            ...state,
+                            part: {
+                                ...state.part,
+                                linnProduced,
+                                sernosSequenceName: 'SERIAL 1',
+                                sernosSequenceDescription: 'MASTER SERIAL NUMBER RECORDS.'
+                            }
+                        };
+                    }
+                    return {
+                        ...state,
+                        part: {
+                            ...state.part,
+                            linnProduced
+                        }
+                    };
+                }
+                if (action.fieldName === 'preferredSupplier') {
+                    return {
+                        ...state,
+                        part: {
+                            ...state.part,
+                            preferredSupplier: action.payload.name,
+                            preferredSupplierName: action.payload.description
+                        }
+                    };
+                }
+                return {
+                    ...state,
+                    part: { ...state.part, [action.fieldName]: action.payload }
+                };
+            default:
+                return state;
+        }
+    }
+
+    // this useReducer call replaces all the old useState calls
+    // we now have one state object and one consistent method of updating it
+    const [state, dispatch] = useReducer(partReducer, {
```

Comment:
Just thinking as well if it would be better just to have state as 'part'? prevPart could still be separate and might make things cleaner? Might also be better with both in the reducer...